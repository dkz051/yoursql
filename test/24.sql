CREATE DATABASE oop;
USE oop;
CREATE TABLE qwe(id INT, stu INT, course CHAR, score INT, PRIMARY KEY(id));
INSERT INTO qwe(id, stu, course, score) VALUES(13, 1001, "LinearAlgebra", 85);
INSERT INTO qwe(id, stu, course, score) VALUES(14, 1002, "LinearAlgebra", 57);
INSERT INTO qwe(id, stu, course, score) VALUES(15, 1003, "LinearAlgebra", 34);
INSERT INTO qwe(id, stu, course, score) VALUES(16, 1004, "LinearAlgebra", 90);
INSERT INTO qwe(id, stu, course, score) VALUES(17, 1005, "LinearAlgebra", 97);
INSERT INTO qwe(id, stu, course, score) VALUES(18, 1006, "LinearAlgebra", 77);
INSERT INTO qwe(id, stu, course, score) VALUES(19, 1007, "LinearAlgebra", 68);
INSERT INTO qwe(id, stu, course, score) VALUES(20, 1010, "LinearAlgebra", 52);
INSERT INTO qwe(id, stu, course, score) VALUES(21, 1011, "LinearAlgebra", 81);
INSERT INTO qwe(id, stu, course, score) VALUES(22, 1012, "LinearAlgebra", 73);
INSERT INTO qwe(id, stu, course, score) VALUES(1, 1001, "Calculus", 100);
INSERT INTO qwe(id, stu, course, score) VALUES(2, 1002, "Calculus", 90);
INSERT INTO qwe(id, stu, course, score) VALUES(3, 1003, "Calculus", 97);
INSERT INTO qwe(id, stu, course, score) VALUES(4, 1004, "Calculus", 47);
INSERT INTO qwe(id, stu, course, score) VALUES(5, 1005, "Calculus", 78);
INSERT INTO qwe(id, stu, course, score) VALUES(6, 1006, "Calculus", 79);
INSERT INTO qwe(id, stu, course, score) VALUES(7, 1007, "Calculus", 85);
INSERT INTO qwe(id, stu, course, score) VALUES(8, 1008, "Calculus", 92);
INSERT INTO qwe(id, stu, course, score) VALUES(9, 1009, "Calculus", 96);
INSERT INTO qwe(id, stu, course, score) VALUES(10, 1010, "Calculus", 99);
INSERT INTO qwe(id, stu, course, score) VALUES(11, 1011, "Calculus", 58);
INSERT INTO qwe(id, stu, course, score) VALUES(12, 1012, "Calculus", 37);
INSERT INTO qwe(id, stu, course, score) VALUES(23, 1001, "OOP", 97);
INSERT INTO qwe(id, stu, course, score) VALUES(24, 1002, "OOP", 56);
INSERT INTO qwe(id, stu, course, score) VALUES(25, 1003, "OOP", 47);
INSERT INTO qwe(id, stu, course, score) VALUES(26, 1004, "OOP", 99);
INSERT INTO qwe(id, stu, course, score) VALUES(27, 1005, "OOP", 100);
INSERT INTO qwe(id, stu, course, score) VALUES(28, 1006, "OOP", 87);
INSERT INTO qwe(id, stu, course, score) VALUES(29, 1007, "OOP", 89);
INSERT INTO qwe(id, stu, course, score) VALUES(30, 1008, "OOP", 76);
INSERT INTO qwe(id, stu, course, score) VALUES(31, 1009, "OOP", 86);
INSERT INTO qwe(id, stu, course, score) VALUES(32, 1010, "OOP", 65);
INSERT INTO qwe(id, stu, course, score) VALUES(33, 1011, "OOP", 86);
INSERT INTO qwe(id, stu, course, score) VALUES(34, 1012, "OOP", 84);
INSERT INTO qwe(id, stu, course, score) VALUES(35, 1013, "OOP", 94);
INSERT INTO qwe(id, stu, course, score) VALUES(36, 1014, "OOP", 61);
SELECT course, COUNT(score), AVG(score), MIN(score), MAX(score), STDDEV(score) FROM qwe GROUP BY course ORDER BY course;
DROP DATABASE oop;
