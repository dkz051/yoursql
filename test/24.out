course	COUNT(score)	AVG(score)	MIN(score)	MAX(score)	STDDEV(score)
Calculus	12	79.8333	37	100	20.41581630882183
LinearAlgebra	10	71.4000	34	97	18.18350901228913
OOP	14	80.5000	47	100	16.31716887208072
